Assets/MRTK/Core/Providers/InputSimulation/SimulatedControllerDataProvider.cs

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+
+/// <summary>
+/// Provides per-frame data access to simulated controller data
+/// </summary>
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Internal class to define current state of a controller.
+    /// </summary>
+    internal abstract class SimulatedControllerState
+    {
+        protected Handedness handedness = Handedness.None;
```

Comment:
I see now the refactoring that you were mentioning around shared stuff between the hand and motion controller data provider!

Reply:
It looks like most of this is pretty much stuff from the existing here:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/ee161fdd81aa6ab591659bae75f64e02c6a2cdce/Assets/MRTK/Core/Providers/InputSimulation/SimulatedHandDataProvider.cs

If there are any interesting deltas, can you call them out?

Reply:
The main difference is all fields and functions that are specific to hand are removed (e.g. `bool IsPinching` and `float gestureBlending` in `SimulatedControllerState` / `jointCount` and `SelectGesture()` in `SimulatedControllerDataProvider`). I think this file (the new `SimulatedHandDataProvider.cs` after shared fields etc. are removed) can provide a clearer picture in terms of deltas:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/a26de109a452639b85011e782666747551c5e746/Assets/MRTK/Core/Providers/InputSimulation/SimulatedHandDataProvider.cs